src/tag-input/src/TagInput.js

```diff
@@ -80,9 +83,11 @@ class TagInput extends React.Component {
   addTags = (value = '') => {
     const { onAdd, onChange, values } = this.props
     const newValues = this.getValues(value)
-    const shouldClearInput =
-      safeInvoke(onAdd, newValues) ||
-      safeInvoke(onChange, values.concat(newValues))
+    let shouldClearInput = safeInvoke(onAdd, newValues)
+
+    if (typeof onChange === 'function') {
```

Comment:
This fixes the `shouldClearInput` bug. `return false` in either `onAdd` OR `onChange` and the input will not clear 👍

Reply:
Sweet! Makes sense.